Still fighting with mingw compilation...

I had to extend the `-I` to `./win32` and `.` in Makefile.

But then, I see a lot of `undefined reference to` ... network related stuff such as __imp_ntohs and others...

--- a/Makefile.in
+++ b/Makefile.in
@@ -33,6 +33,12 @@ INSTALL=install
 MKDIR=mkdir -p
 OS := $(shell uname -o)
 
+ifndef OS
+# This could happen if the Makefile is unable to run "uname", which can
+# happen if the shell has a bad path (or is the wrong shell)
+$(error Could not determine the OS, cannot continue)
+endif
+
 INSTALL_PROG=$(INSTALL) -m755
 INSTALL_DOC=$(INSTALL) -m644
 

--- a/scripts/hack_fakeautoconf
+++ b/scripts/hack_fakeautoconf
@@ -19,16 +19,8 @@ cat Makefile.in | sed \
     -e "s%@N2N_LIBS@%$LDLIBS%g" \
     > Makefile
 
-if [ "$RUNNER_OS" = "Windows" ]; then
-    CFLAGS="$CFLAGS -I.. -I../win32"
-    #LDLIBS inherited from above
-fi
-
 cat tools/Makefile.in | sed \
     -e "s%@CC@%gcc%g" \
-    -e "s%@N2N_LIBS@%$LDLIBS%g" \
-    -e "s%@CFLAGS@%$CFLAGS%g" \
-    -e "s%@LDFLAGS@%$LDFLAGS%g" \
     -e "s%@ADDITIONAL_TOOLS@%%g" \
     > tools/Makefile
 

--- a/tools/Makefile.in
+++ b/tools/Makefile.in
@@ -1,6 +1,8 @@
 CC=@CC@
 DEBUG?=-g3
 
+OS:=$(shell uname -o)
+
 INSTALL=install
 INSTALL_PROG=$(INSTALL) -m755
 MKDIR=mkdir -p
@@ -14,8 +16,11 @@ endif
 
 HEADERS=$(wildcard include/*.h)
 CFLAGS+=-I../include
-LDFLAGS+=-L..
+ifeq ($(OS),Msys)
+CFLAGS+=-I../win32
+endif
 CFLAGS+=$(DEBUG)
+LDFLAGS+=-L..
 
 N2N_LIB=../libn2n.a
 

